specs/~dynamic-names.yml

```
@@ -0,0 +1,349 @@
+overview: |
+  Rationale: this special notation was introduced primarily to allow the dynamic
+  loading of partials. The main advantage that this notation offers is to allow
+  dynamic loading of partials, which is particularly useful in cases where
+  polymorphic data needs to be rendered in different ways, cases which would
+  otherwise be possible to render only with solutions that are convoluted,
+  inefficient, or both.
+  Example.
+  Let's consider the following data:
+    items: [
+      { content: 'Hello, World!' },
+      { url: 'http://example.com/foo.jpg' },
+      { content: 'Some text' },
+      { content: 'Some other text' },
+      { url: 'http://example.com/bar.jpg' },
+      { url: 'http://example.com/baz.jpg' },
+      { content: 'Last text here' }
+    ]
+  The goal is to render the different types of items in different ways: the
+  items having a key named `content` should be rendered with the template
+  `text.mustache` and the items having a key named `url` should be rendered
+  with the template `image.mustache`:
+  text.mustache:
+    {{!image.mustache}}
+    <img src="{{url}}"/>
+  image.mustache:
+    {{!text.mustache}}
+    {{content}}
+  There are already several ways to achieve this goal, here below are
+  illustrated and discussed the most significant solutions to this problem.
+  ## Using Pre-Processing
```

If YAML interprets the hashes as starting a comment, you can also use Markdown's underline style of indicating headers:

```suggestion
  Using Pre-Processing
  --------------------
```

---

I think it's better if we just remove the leading hashtags.

---

Fine by me.